ui/app/decorators/unload-model-route.js

```diff
@@ -0,0 +1,48 @@
+import { inject as service } from '@ember/service';
+import { removeModel } from '../utils/remove-record';
+
+/**
+ * Takes array of model names which should be unloaded when moving away from
+ * the route. Automatically unloads all `capabilities` and `control-group`
+ * data before unloading the specified models.
+ *
+ *** basic example
+ *
+ * import Route from '@ember/route';
+ * import withUnloadModelRoute from 'vault/decorators/unload-model-route';
+ *
+ * @withUnloadModelRoute(['policy/acl'])
+ * class PolicyViewRoute extends Route { foo = null; }
+ *
+ */
+
+export function withUnloadModelRoute(models) {
+  return function decorator(SuperClass) {
+    return class UnloadModelRoute extends SuperClass {
+      @service store;
+
+      static _models;
+
+      constructor() {
+        super(...arguments);
+        if (!models || !Array.isArray(models)) {
+          throw new Error('models must be an array of model types as strings');
```

Comment:
Not sure how specific we want to get but could also check that the classes passed in the models array inherit `Model`:

```js
Object.prototype.isPrototypeOf.call(Model, SuperClass)
```